lib/sequel/database/transactions.rb

```diff
@@ -198,6 +198,8 @@ def _trans(conn)
     def add_transaction(conn, opts)
       hash = {}
 
+      apply_transaction_options(conn, opts, hash)
```

Comment:
I think it would be better to change this method so that it returned the hash to use (possible name: `transaction_state_hash`), instead of relying on side-effects.